python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/sensor_builder.py

```diff
@@ -0,0 +1,264 @@
+    materializations_and_timestamps = [
+        (mat.metadata[COMPLETED_AT_TIMESTAMP_METADATA_KEY].value, mat) for mat in asset_events
+    ]
+    return [
+        sorted_event[1]
+        for sorted_event in sorted(
+            materializations_and_timestamps, key=lambda x: (x[0], topo_aks.index(x[1].asset_key))
+        )
+    ]
+
+
+@preview
+def build_dbt_cloud_polling_sensor(
+    *,
+    workspace: DbtCloudWorkspace,
+    dagster_dbt_translator: Optional[DagsterDbtTranslator] = None,
+    minimum_interval_seconds: int = DEFAULT_DBT_CLOUD_SENSOR_INTERVAL_SECONDS,
+    default_sensor_status: Optional[DefaultSensorStatus] = None,
+) -> SensorDefinition:
+    """The constructed sensor polls the dbt Cloud Workspace for activity, and inserts asset events into Dagster's event log.
+
+    Args:
+        workspace (DbtCloudWorkspace): The dbt Cloud workspace to poll for runs.
+        dagster_dbt_translator (Optional[DagsterDbtTranslator], optional): The translator to use
+            to convert dbt Cloud content into :py:class:`dagster.AssetSpec`.
+            Defaults to :py:class:`DagsterDbtTranslator`.
+        minimum_interval_seconds (int, optional): The minimum interval in seconds between sensor runs. Defaults to 30.
+        default_sensor_status (Optional[DefaultSensorStatus], optional): The default status of the sensor.
+
+    Returns:
+        Definitions: A `Definitions` object containing the constructed sensor.
+    """
+    dagster_dbt_translator = dagster_dbt_translator or DagsterDbtTranslator()
+
+    @sensor(
+        name=f"{workspace.account_name}_{workspace.project_name}_{workspace.environment_name}__run_status_sensor",
+        minimum_interval_seconds=minimum_interval_seconds,
+        default_status=default_sensor_status or DefaultSensorStatus.RUNNING,
+        # This sensor will only ever execute asset checks and not asset materializations.
+        asset_selection=AssetSelection.all_asset_checks(),
+    )
+    def dbt_cloud_run_sensor(context: SensorEvaluationContext) -> SensorResult:
+        """Sensor to report materialization events for each asset as new runs come in."""
+        context.log.info(
+            f"************"
+            f"Running sensor for dbt Cloud wokspace for account {workspace.account_name}, "
+            f"project {workspace.project_name} and environment {workspace.environment_name}"
+            f"***********"
+        )
+        try:
+            cursor = (
+                deserialize_value(context.cursor, DbtCloudPollingSensorCursor)
+                if context.cursor
+                else DbtCloudPollingSensorCursor()
+            )
+        except Exception as e:
+            context.log.info(f"Failed to interpret cursor. Starting from scratch. Error: {e}")
+            cursor = DbtCloudPollingSensorCursor()
+        current_date = get_current_datetime()
+        current_offset = cursor.offset or 0
+        finished_at_start = (
```

Comment:
nitty but I find finished_at_start pretty confusing to look at. Maybe finished_at_upper_bound and finished_at_lower_bound